src/infra/core/src/services/ingest/push_ingest_executor_impl.rs

```diff
@@ -206,6 +206,13 @@ impl PushIngestExecutorImpl {
                     _ => Err(format!("Unsupported source: {url}").int_err().into()),
                 }
             }
+            DataSource::Buffer(buf) => {
+                // Save buffer to a file so datafusion can read it
+                // TODO: We likely can avoid this and build a DataFrame directly
+                tracing::info!(path = ?temp_path, "Copying buffer into a temp file");
+                tokio::fs::write(&temp_path, buf).await.int_err()?;
```

Comment:
Q: Please remind me whether `args.operation_dir` will be cleared after a successful `do_ingest()` iteration?

Since `temp_path` is technically not a temporary folder that will delete itself

Reply:
Currently we clear it on every run of CLI tool. We don't clear it in node yet and will need to implement a GC for that.